src/core/HttpResponse.ts

```diff
@@ -35,10 +33,15 @@ export interface StrictResponse<BodyType extends DefaultBodyType>
  *
  * @see {@link https://mswjs.io/docs/api/http-response `HttpResponse` API reference}
  */
-export class HttpResponse extends Response {
-  constructor(body?: BodyInit | null, init?: HttpResponseInit) {
+export class HttpResponse<BodyType extends DefaultBodyType>
```

Comment:
The trick is this:

1. `implements StrictResponse<T>`
2. All static methods return `HttpRespone<T>` now, _not_ `StrictResponse`.
3. The response resolver type is annotated to return `HttpResponse<T>` as well. 

This makes responses constructed and created via static methods to have the same type validation—on the body argument type level, not the return type level of the response resolver.